client/src/ui/BitbakeCommands.ts

```diff
@@ -429,6 +431,28 @@ async function devtoolOpenWorkspaceCommand (bitbakeWorkspace: BitbakeWorkspace,
   await vscode.commands.executeCommand('vscode.openFolder', vscode.Uri.file(workspacePath), { forceNewWindow: true })
 }
 
+async function openRecipeWorkdirCommand (bitbakeWorkspace: BitbakeWorkspace, bitBakeProjectScanner: BitBakeProjectScanner, client: LanguageClient, uri?: any): Promise<void> {
+  const chosenRecipe = await selectRecipe(bitbakeWorkspace, bitBakeProjectScanner, uri)
+  if (chosenRecipe === undefined) { return }
+
+  logger.debug(`Command: open-recipe-workdir: ${chosenRecipe}`)
+  const recipeUri = bitBakeProjectScanner.scanResult._recipes.find((recipe) => recipe.name === chosenRecipe)?.path
+  if (recipeUri === undefined) throw Error(`Recipe ${chosenRecipe}: not found in Bitbake scan`)
+  const recipePath = pathInfoToString(recipeUri)
+  let recipeWorkdir: string | undefined | null = await client.sendRequest('bitbake/getVar', { variable: 'WORKDIR', recipe: chosenRecipe })
```

Comment:
There is already a notification when the scan is triggered.
I prefer this implementation over ProcessRecipeScanResults because it's refactored into a common function with the ability to scan and get results from the client in an optimal way. It should work for browse dir, _task_deps and future ideas we have. It doesn't scan when not needed but allows doing it if needed. Meanwhile, it doesn't need to store values on the client which is more maintainable and will be as fast if the scan is already in the server's memory.